internal/rpc/clients.go

```diff
@@ -0,0 +1,123 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package rpc
+
+import (
+	"crypto/tls"
+	"fmt"
+	"net/http"
+	"time"
+
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+	"google.golang.org/grpc"
+	"open-match.dev/open-match/internal/config"
+)
+
+var (
+	clientLogger = logrus.WithFields(logrus.Fields{
+		"app":       "openmatch",
+		"component": "client",
+	})
+)
+
+// ClientParams contains the connection parameters to connect to an Open Match service.
+type ClientParams struct {
+	Hostname       string
+	Port           int
+	TrustedKeyPath string
```

Comment:
Shouldn't this be `TrustedCertificate []byte`
The reason why the file data (in PEM format) is specified is that there's 2 paths to load from, a file or the PEM data. They both converge on the PEM data so it make sense of the code flow join to happen there.